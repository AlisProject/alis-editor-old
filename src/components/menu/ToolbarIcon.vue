<template>
  <div class="editor-toolbar__icon" :class="{ 'is-active': active }">
    <div class="editor-toolbar__iconContent" v-html="src" />
  </div>
</template>

<script lang="ts">
export default {
  props: {
    active: Boolean,
    src: String
  }
}
</script>

<style>
.editor-toolbar__icon {
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  transition: all 0.15s ease-out;
}

.editor-toolbar__icon .fill-target {
  fill: #5e5e5e;
}

.editor-toolbar__icon.is-active {
  background: #0086cc;
}

.editor-toolbar__icon.is-active .fill-target {
  fill: #fff;
}
</style>
