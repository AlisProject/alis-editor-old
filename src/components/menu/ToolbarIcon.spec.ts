import { mount } from '@vue/test-utils'
import ToolbarIcon from './ToolbarIcon.vue'

describe('ToolbarIcon.vue', () => {
  test('renders correctly', () => {
    const wrapper = mount(ToolbarIcon)
    expect(wrapper.element).toMatchSnapshot()
  })
})
