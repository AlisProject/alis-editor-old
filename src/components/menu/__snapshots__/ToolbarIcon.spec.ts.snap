// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarIcon.vue renders correctly 1`] = `
<div
  class="editor-toolbar__icon"
>
  <div
    class="editor-toolbar__iconContent"
  />
</div>
`;
